src/allowed-blocks-setup.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import { map, without } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { dispatch } from '@wordpress/data';
+import { getBlockTypes } from '@wordpress/blocks';
+/**
+ * Internal dependencies
+ */
+import { JETPACK_DATA_PATH } from '../jetpack/projects/plugins/jetpack/extensions/shared/get-jetpack-data';
+import { registerBlock as registerJetpackLayoutGridBlock } from '../block-experiments/blocks/layout-grid/src';
+import {
+	registerContactInfoBlock as registerJetpackContactInfoBlock,
+	registerStoryBlock as registerJetpackStoryBlock,
+} from '../jetpack/projects/plugins/jetpack/extensions/editor.native';
+
+// Please also consider updating ./block-support/supported-blocks.json
+const availableJetpackBlocks = {
+	'contact-info': { available: true },
+	'layout-grid': { available: true },
+	story: { available: true },
+};
+
+const setJetpackData = ( {
+	isJetpackActive = false,
+	userData: tracksUserData = null,
+	siteFragment = null,
+	blogId: wpcomBlogId = 1,
+} ) => {
+	return {
+		siteFragment,
+		tracksUserData,
+		wpcomBlogId,
+		jetpack: { is_active: isJetpackActive },
+		available_blocks: availableJetpackBlocks,
+	};
+};
+
+const registerJetpackBlocksIfCapable = ( props = {} ) => {
+	const {
+		capabilities: {
+			layoutGridBlock = false,
+			mediaFilesCollectionBlock = false,
+			contactInfoBlock = false,
+		} = {},
+	} = props;
+
+	if ( layoutGridBlock ) {
+		registerJetpackLayoutGridBlock();
+	}
+
+	if ( mediaFilesCollectionBlock ) {
+		registerJetpackStoryBlock();
+	}
+
+	if ( contactInfoBlock ) {
+		registerJetpackContactInfoBlock();
+	}
```

Comment:
I noticed that originally we were hiding these blocks instead of preventing registering them:
https://github.com/wordpress-mobile/gutenberg-mobile/blob/6fea169a3c04e6fa441c5369d60539df2a407ca9/src/jetpack-editor-setup.js#L68-L69
https://github.com/wordpress-mobile/gutenberg-mobile/blob/6fea169a3c04e6fa441c5369d60539df2a407ca9/src/jetpack-editor-setup.js#L47-L53

From my POV, I think it makes sense to register them conditionally to the props but I'm wondering if there was a reason to do it that way.

@illusaen do you have any insights regarding this topic?